datasets/us/doh_debarred/crawler.py

```diff
@@ -0,0 +1,71 @@
+from zavod import Context
+import csv
+from zavod import helpers as h
+from typing import Dict, Any
+
+FORMATS = ["%Y%m%d"]
+def get_first_line(row: Dict[str, Any]):
+    address = row.pop("ADDRESS")
+    if address:
+        address_data = address
+    else:
+        address_data = row.get("ZIP")  # Use ZIP code as ID for empty address
+    return address_data
+
+def crawl_item(context: Context, row: Dict[str, Any]):
+    if row["LASTNAME"].strip() or row["FIRSTNAME"].strip():
+        entity = context.make("Person")
+        full_name=h.make_name(
+        first_name = row.pop("FIRSTNAME",None),
+        middle_name = row.pop("MIDNAME",None),
+        last_name = row.pop("LASTNAME",None)
+        )
+        
+        context.log.info(full_name)
+        address_data = get_first_line(row)
+        entity.id = context.make_slug(full_name, address_data)
+        h.apply_name(entity=entity,full=full_name,lang='eng')
+        entity.add("birthDate", h.parse_date(row.pop("DOB",None), FORMATS))
+        entity.add("position",f'{row.pop("GENERAL",None)} {row.pop("SPECIALTY",None)}')
+    else:
+        entity = context.make("Company")
+        name = row.pop("BUSNAME",None)
+        address_data = get_first_line(row)
+        entity.id = context.make_slug(name, address_data)
+        entity.add("name", name)
+        entity.add("description",f'{row.pop("GENERAL",None)} {row.pop("SPECIALTY",None)}')
+   
+    address=h.make_address(
+        context,
+        city=row.pop("CITY",None),
+        state=row.pop("STATE",None),
+        postal_code=row.pop("ZIP",None),
+        country_code="us"
+    )
+    h.apply_address(context,entity,address)
+    h.make_identification(
+        context,
+        entity,
+        key=row.pop("UPIN",None),
+        number=row.pop("NPI",None),
+    )
```

Comment:
no, for each of these two types, we need a distict identification entity, which indicates its type in the idenfication type property.